templates/templates/preview/model_plain_text.template

```diff
@@ -0,0 +1,97 @@
+                {{ model.path }}
+                {% endfor %}
+            {%- endif %}
+            {%- if model.files.link %}
+            Symlink: 
+                {%- for link in model.files.link %}
+                {{ link }} ->
+                {{ model.path }}
+                {% endfor %}
+            {%- endif %}
+            {%- endif %}
+            {%- if model.run_settings.colocated_db_settings %}
+            Colocated:
+            {%- if model.run_settings.colocated_db_settings.db_identifier %}
+                Database identifier: {{ model.run_settings.colocated_db_settings.db_identifier }}
+            {% endif %}
+                {%- if model.run_settings.colocated_db_settings.port %}
+                Connection Type: TCP
+                TCP/IP Port:
+                    {{ model.run_settings.colocated_db_settings.port }}
+                {% endif -%}
+                {%- if model.run_settings.colocated_db_settings.unix_socket %}
+                Connection Type: UDS
+                Unix socket: {{ model.run_settings.colocated_db_settings.unix_socket }}
+                {%- endif -%}
+                Ifname: {{ model.run_settings.colocated_db_settings.ifname[0] }}
+                Cpus: {{ model.run_settings.colocated_db_settings.cpus }}
+                Custom pinning: {{ model.run_settings.colocated_db_settings.custom_pinning }}
+            {%- if model.run_settings.colocated_db_settings['db_scripts'] %}
+            Scripts:
+            {%- for script in model.run_settings.colocated_db_settings['db_scripts'] %}   
+                
+                Script name: {{ script.name }}
+                Script path: {{ script.script_path }}
+                Backend: {{ script.device }}   
+                Devices per node: {{ script.devices_per_node }} 
+            {%- endfor %}
+            {%- endif %}
+            {%- if model.run_settings.colocated_db_settings['db_models'] %}
+            Models: 
+            {%- for mlmodel in model.run_settings.colocated_db_settings['db_models'] %}
+                
+                Model name: {{ mlmodel.name }}
+                Backend: {{ mlmodel.backend }} 
+                Model path: {{ mlmodel.model_file }} 
+                Device: {{ mlmodel.device }}  
+                Devices per node: {{ mlmodel.devices_per_node }}
+                First device:{{ mlmodel.first_device }}
+                {%- for input in mlmodel.inputs %}
+                Inputs: 
+                    {{ input }} 
+                {%- endfor %}
+                {%- for output in mlmodel.outputs %}
+                Outputs: 
+                    {{ output }}
+                {%- endfor %}
+            {%- endfor %}
+            {%- endif %}
+            {%- endif %}
+            {%- if manifest.models %}
+            {%- if manifest.models[0].query_key_prefixing() == True %}  
+            Key prefix: {{ model.name }}
+            {%- endif %}
+            {%- endif %}
+    {%- endif -%}
```

Comment:
nitpick: our if/endif wrapping the whole thing don't have the same indentation. could we fix the endif?

maybe also use jinja to get rid of whitespace and shift the first line down off the if clause just for readability?

Reply:
changed